networking_cisco/plugins/cisco/db/l3/l3_router_appliance_db.py

```diff
@@ -453,6 +450,16 @@ def add_router_interface(self, context, router_id, interface_info):
         routers = [self.get_router(context, router_id)]
         self.add_type_and_hosting_device_info(context.elevated(), routers[0])
         if driver:
+            if by_subnet:
```

Comment:
I think the problem is the variable name (port_db). The object here is actually a dictionary, and not the database object itself. As a result, the port needs to be updated from the precommit, since the port object in the DB has been updated (in this case, the device_id has changed to point to the router that owns it). I can change the variable name to get rid of the _db suffix.